src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -352,14 +352,22 @@ namespace Microsoft.FSharp.Core
     [<Sealed>]
     type RequireQualifiedAccessAttribute() =
         inherit System.Attribute()
-      
+
     [<AttributeUsage (AttributeTargets.Class ||| AttributeTargets.Assembly,AllowMultiple=true)>]  
     [<Sealed>]
     type AutoOpenAttribute(path:string) =
         inherit System.Attribute()
         member x.Path = path
         new() =  AutoOpenAttribute("")
 
+    /// This Attribute is used to make Value bindings like
+    ///      let x = some code
+    /// operate like static properties.
+    [<AttributeUsage(AttributeTargets.Property,AllowMultiple=false)>]
+    [<Sealed>]
+    type ValueAsStaticPropertyAttribute() =
```

Comment:
I'm not totally in love with the name since all non-function values are compiled as static properties - this is really suppressing the field for the value and making it call-by-need. But it will do for now since it is internal and fslib only.